core/commands/config.go

```diff
@@ -140,11 +148,29 @@ Set the value of the 'Datastore.Path' key:
 	Type: ConfigField{},
 }
 
+// Returns bool to indicate if tested key is in the blocked scope.
+// (scope includes parent, direct, and child match)
+func inBlockedScope(testKey string, blockedScope string) bool {
+	blockedScope = strings.ToLower(blockedScope)
+	roots := strings.Split(strings.ToLower(testKey), ".")
+	var scope []string
+	for _, name := range roots {
+		scope := append(scope, name)
+		impactedKey := strings.Join(scope, ".")
+		// blockedScope=foo.bar.BLOCKED should return true
+		// for parent and child impactedKeys: foo.bar and foo.bar.BLOCKED.subkey
+		if strings.HasPrefix(impactedKey, blockedScope) || strings.HasPrefix(blockedScope, impactedKey) {
```

Comment:
This || has the effect of blocking every step along the bockedScope path. Is that really what is intended?

Reply:
@petar this is a quick fix to avoid leaking `Api.Key` via `ipfs config Pinning`
Ok for now, but we will need smarter sanitization when you add MFS repining fields.